lib/permissions.js

```diff
@@ -0,0 +1,402 @@
+'use strict'
+/**
+ * Provides a Solid web client module to get and set authorization rules for a resource.
+ * @module permissions
+ */
+var PermissionType = {
+  Owner: 'Owner',
+  Read: 'Read',
+  Write: 'Write',
+  Append: 'Append'
+}
+
+var vocab = require('./vocab')
+var rdf = require('./util/rdf-parser').rdflib
+var auth = require('./auth')
+var webClient = require('./web')
+
+/**
+ * Set and Get authorization rules for a resource
+ * @class Permissions
+ * @static
+ */
+var Permissions = {
+  /**
+   * Takes a resource URI and a list of agents & permissions,
+   * creates and acl resource corresponding to the given resource URI.
+   * @method setPermissions
+   * @param url {String} Resource URI to set permissions for
+   * @param agents {[String]} List of agent WebIDs to set the permissions for
+   * @param permissions {[PermissionType]} List of permissions
+   * @return {Promise(String)} acl resource URI
+   */
+  setPermissions: function setPermissions (url, agents, permissions) {
+    return new Promise(function (resolve, reject) {
+      // reading the existing acls from the server and merging them with the new acls for each agent
+      mergeAcls(url, agents, permissions).then(
+        function (mergedPermissions) {
+          // having the merged permissions, create the RDF string representing the auth rules
+          // for the list of agents
+          var aclString = createAclRules(url, agents, permissions)
+          aclString = appendOwnerRule(aclString, url)
+          // write the new acl resource via put http method to replace any existing one
+          webClient.put(url + ',acl', aclString).then(
```

Comment:
Right. @dmitrizagidulin just pointed me to this. I guess we need to find a consistent way to identify how the naming of acl resources go on the server or at least the pattern (specs) that shows how the naming should be.